presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1011,6 +1064,14 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
                 return createAndAssignScope(table, scope, outputFields);
             }
 
+            Optional<ConnectorMaterializedViewDefinition> optionalMaterializedView = metadata.getMaterializedView(session, name);
```

Comment:
We need to add a TODO here right? This is where the partition stitching happens. Without TODO, it's very easy to miss this part.